jung-algorithms/src/main/java/edu/uci/ics/jung/layout/model/PointModel.java

```diff
@@ -122,4 +105,39 @@
    * @param z
    */
   void offset(P p, double x, double y, double z);
+
+  /**
+   * compute and return the square of the distance between 2 points
+   *
+   * @param from
+   * @param to
+   * @return squared distance between points
+   */
+  default double distanceSquared(P from, P to) {
+    double deltaX = getX(to) - getX(from);
+    double deltaY = getY(to) - getY(from);
+    double deltaZ = getZ(to) - getZ(from);
+    return deltaX * deltaX + deltaY * deltaY + deltaZ * deltaZ;
+  }
+
+  /**
+   * compute and return the distance between 2 points
+   *
+   * @param from
+   * @param to
+   * @return distance between points
+   */
+  default double distance(P from, P to) {
+    return Math.sqrt(distanceSquared(from, to));
+  }
+
+  /**
+   * computes the distance between p and the origin
+   *
+   * @param p the point to consider
+   * @return the distance to the origin
+   */
+  default double distance(P p) {
```

Comment:
There is no 'this' state in PointModel implementations. Maybe you are looking at things in isolation without considering the big picture.

Reply:
Yeah, I probably am.  :)  (I could really use a design doc, or at least an architectural diagram.)

Which leads to a question: why did you design `PointModel` as (in essence) a collection of stateless (i.e., could be static) methods, rather than something like `Point2D` (in which an object represents a location)?  It's not a very OO way of doing things.  I'm assuming that you had a good reason, but I'm not seeing it.

Reply:
I wanted to use something in the LayoutAlgorithms that was not dependent on a specific library but was not a 'yet-another-point-class' to copy values to and from.

Reply:
If they were static methods, i could not override them.